uint/src/uint.rs

```diff
@@ -1051,20 +1152,51 @@ macro_rules! construct_uint {
 			}
 		}
 
-		impl ::core::ops::Mul<$name> for $name {
+		impl<'a> ::core::ops::Mul<u32> for &'a $name {
 			type Output = $name;
 
-			fn mul(self, other: $name) -> $name {
-				let (result, overflow) = self.overflowing_mul(other);
-				panic_on_overflow!(overflow);
-				result
+			fn mul(self, other: u32) -> $name {
+				*self * other
 			}
 		}
 
-		impl ::core::ops::Div<$name> for $name {
+		impl ::core::ops::MulAssign<u32> for $name {
+			fn mul_assign(&mut self, other: u32) {
+				let result = *self * other;
+				*self = result
+			}
+		}
+
+		// all other impls
+		impl_mul_from!($name, u8);
+		impl_mul_from!($name, u16);
+		impl_mul_from!($name, u64);
+		impl_mul_from!($name, usize);
+
+		impl_mul_from!($name, i8);
+		impl_mul_from!($name, i16);
+		impl_mul_from!($name, i64);
+		impl_mul_from!($name, isize);
+
+		impl_mul_from!($name, $name);
+
+		impl_mulassign_from!($name, u8);
+		impl_mulassign_from!($name, u16);
+		impl_mulassign_from!($name, u64);
+		impl_mulassign_from!($name, usize);
+
+		impl_mulassign_from!($name, i8);
```

Comment:
Seems very confusing to me, we don't really support signed INT256, so what exactly does this do in case of negative numbers?

Reply:
we panic if integers < 0 is passed in, but the impls for signed numbers are there as rust defaults number types to i32 I think, as well as to be consistent with the other impls of these methods which take an Into<uint>

Reply:
Makes sense! Thanks for the explanation.